### ISSUE TITLE
bun build --compile doesn't work correctly in a dev container

### ISSUE THREAD

**Issue by krisalyssa** [oven-sh/bun#12318]
### What version of Bun is running?

1.1.17

### What platform is your computer?

Linux 6.6.26-linuxkit aarch64 unknown

### What steps can reproduce the bug?

(running in a dev container based on https://github.com/marcosgomesneto/bun-devcontainers/pkgs/container/bun-devcontainers%2Fbasic-bun in Docker Desktop 4.30.0 (149282) on macOS 14.5 Apple Silicon)

```shell
root@1a44d0a1ad62:/workspaces/bun# echo $(cat cli.ts)
console.log("Hello world!");
root@1a44d0a1ad62:/workspaces/bun# bun build ./cli.ts --compile --outfile mycli
  [10ms]  bundle  1 modules
error: failed to rename /run/host_virtiofs/Users/kcottingham/Projects/bun/.17debdb779bfefff-00000000.bun-build to mycli: ENOENT
root@1a44d0a1ad62:/workspaces/bun# pwd
/workspaces/bun
root@1a44d0a1ad62:/workspaces/bun# mv ./.17debdb779bfefff-00000000.bun-build mycli
root@1a44d0a1ad62:/workspaces/bun# chmod +x mycli 
root@1a44d0a1ad62:/workspaces/bun# ./mycli
Hello world!
```

### What is the expected behavior?

The compiled binary runs and outputs `Hello world!`.

### What do you see instead?

`bun build --compile` fails to rename the temporary output file to `mycli`.

### Additional information

`bun build --compile` is building the path to the temporary output file incorrectly.

**Comment by axelson** [comment]
@krisalyssa did you figure out a workaround to this? I'm running into the same issue.

**Comment by krisalyssa** [comment]
@axelson No. I was evaluating `bun` for use in our organization and for this and other reasons we're not continuing with it.

**Comment by axelson** [comment]
Okay thanks. Yeah, same here, or at least minimizing our use of it.

**Comment by svefro** [comment]
Same issue here. Want to compare bun exe size vs deno.

**Comment by mariocao** [comment]
Same here 😭

**Comment by Thomasvdam** [comment]
In the end @mariocao and I found a workaround. We run the build command in the `/tmp` directory of the devcontainer and have the output file also be written to the `/tmp` directory. Then we manually copy the artefact from the `/tmp/artefact_name` location to the desired location in the workspace (since Bun build tries to do a rename which doesn't work cross device 🤕).

**Comment by cjost1988** [comment]
Hi, 

I also stumbled over this in a devcontainer in VSCode and tried it also directly with the `oven-sh/bun:latest` docker image and ran the `--compile` command. It also fails.

```bash
❯ docker run -it -v ./backend:/home/bun/app oven/bun:latest  build ./index.ts --compile --outfile server                                             
   [9ms]  bundle  1 modules
failed to rename /run/host_virtiofs/Users/<host-user>/<project>/backend/.186a29ff9dbf7dbf-00000000.bun-build to server: ENOENT
```

But when I build my own docker image and copy my sources into the container and execute it within the docker image creation, it results with a compiled binary.

```
FROM oven/bun:latest

WORKDIR /app

COPY ./backend /app/

RUN bun build ./index.ts --compile --outfile server

CMD [ "bash" ]
```
